clientconn.go

```diff
@@ -736,6 +736,7 @@ func (ac *addrConn) connect() error {
 		return nil
 	}
 	ac.mu.Unlock()
+	ac.updateConnectivityState(connectivity.Connecting)
```

Comment:
Add some comment so we don't delete it again later.

Reply:
Done. And we want the update *before* unlocking, not after.